Fluid/Ast/ForStatement.cs

```diff
@@ -95,9 +110,10 @@ public override async Task<Completion> WriteToAsync(TextWriter writer, TextEncod
 
             try
             {
-                var forloop = new Dictionary<string, FluidValue>();
-                forloop.Add("length", new NumberValue(list.Count));
-                context.SetValue("forloop", new DictionaryValue(new FluidValueDictionaryFluidIndexable(forloop)));
+                var forloop = new ForLoopValue();
+
+                forloop.Length = list.Count;
+                context.SetValue("forloop", forloop);
 
                 for (var i = 0; i < list.Count; i++)
```

Comment:
nit: list.Count is accessed many times and it requires lookup to Count as compiler doesn't know if it's changing